src/inmanta/data/__init__.py

```diff
@@ -4031,6 +4093,153 @@ def to_dict(self) -> JsonType:
         return dict_result
 
 
+class ResourceOrder(DatabaseOrder):
+    """ Represents the ordering by which resources should be sorted"""
+
+    validator_dataclass = Resource
+    valid_sort_columns = {"resource_type": str, "agent": str, "status": str, "resource_id_value": str}
+
+    @classmethod
+    def parse_from_string(
+        cls,
+        sort: str,
+    ) -> "ResourceOrder":
+        database_order = super().parse_from_string(sort)
+        return ResourceOrder(database_order.order_by_column, database_order.order)
+
+    def get_order_by_db_column_name(self) -> ColumnNameStr:
+        return ColumnNameStr(
+            f"{super().get_order_by_db_column_name()}{'::text' if self._should_be_treated_as_string() else ''}"
+        )
+
+    def _should_be_treated_as_string(self) -> bool:
+        """Ensure that records are sorted alphabetically by status instead of the enum order"""
+        return self.order_by_column == "status"
+
+
+class ResourceHistoryOrder(DatabaseOrder):
+    """ Represents the ordering by which resource history should be sorted"""
+
+    # Sorting based on the date of the configuration model
+    validator_dataclass = ConfigurationModel
+    valid_sort_columns = {"date": datetime.datetime}
+
+    @classmethod
+    def parse_from_string(
+        cls,
+        sort: str,
+    ) -> "ResourceHistoryOrder":
+        database_order = super().parse_from_string(sort)
+        return ResourceHistoryOrder(database_order.order_by_column, database_order.order)
+
+
+class ResourceLogOrder(DatabaseOrder):
+    """ Represents the ordering by which resource logs should be sorted"""
+
+    validator_dataclass = ResourceAction
+    valid_sort_columns = {"timestamp": datetime.datetime}
+
+    @classmethod
+    def parse_from_string(
+        cls,
+        sort: str,
+    ) -> "ResourceLogOrder":
+        database_order = super().parse_from_string(sort)
+        return ResourceLogOrder(database_order.order_by_column, database_order.order)
+
+
+class CompileReportOrder(DatabaseOrder):
+    """ Represents the ordering by which compile reports should be sorted"""
+
+    validator_dataclass = Compile
+    valid_sort_columns = {"requested": datetime.datetime}
+
+    @classmethod
+    def parse_from_string(
+        cls,
+        sort: str,
+    ) -> "CompileReportOrder":
+        database_order = super().parse_from_string(sort)
+        return CompileReportOrder(database_order.order_by_column, database_order.order)
+
+
+class QueryBuilder:
```

Comment:
For me the first issues for reviewing this is the total lack of any comments. This is a helper class so that means that:

- Its purpose should be documented
- Every single method needs documentation on its arguments and return values and preferably a single sentence description of its purpose.

Reply:
This data module/package is already a big mess I am aware of this. However, when we added helpers this should at least be at the top of the file so that the structure is maintained. Even better would be adding it to a helpers.py

This looks like a small issue, but once a file reaches this size just the size alone makes it complex to review.